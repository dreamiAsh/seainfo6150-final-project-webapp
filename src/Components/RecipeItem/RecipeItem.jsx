import React from "react";
import styles from "./RecipeItem.module.css";
import { Link, BrowserRouter } from "react-router-dom";

const RecipeItem = (props) => {
  return (
    <BrowserRouter forceRefresh={true}><Link to={`/recipe/${props.recipe.title}/${props.recipe.imageId}/${props.recipe.author}/${props.recipe.authorInfo}/${props.recipe.reviews}/${props.recipe.likes}/${props.recipe.shortText}/${props.recipe.ingredient}/${props.recipe.step1}/${props.recipe.step2}`}  className={styles.box}>
        <div className={styles.imgBox}>
          <img src={props.recipe.image} alt="recipe image"  className={styles.img}/>
        </div>
        <p className={styles.title}>{props.recipe.title}</p>
        <p className={styles.by}>By <span  className={styles.author}>{props.recipe.author}</span></p>
    </Link></BrowserRouter>
  );
};

export default RecipeItem;
