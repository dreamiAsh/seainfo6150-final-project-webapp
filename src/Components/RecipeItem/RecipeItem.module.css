.box{
    width: 25%;
    height: 210px;
    overflow: hidden;
    float: left;
    margin: 0 4% 24px 4%;
    border: solid 1px #eee;
    text-align: left;
    text-decoration: none;
    color: black;
}
.box:hover{
    border: solid 1px black;
}
.title{
    font-weight: bold;
    font-size: 13px;
    margin: 7px 13px;
}
.by{
    color: gray;
    margin-left: 13px;
    font-size: 10px;
}
.author{
    color: black;
    font-size: 10px;
}
.imgBox{
    width: 100%;
    overflow: hidden;
}
.img{
    width: 100%;
}

@media (min-width: 600px) {
    .title {
        font-size: 15px;
    }
    .by{
        font-size: 12px;
    }
    .author{
        font-size: 15px;
    }
    .box{
        height: 250px;
    }
}
@media (min-width: 800px) {
    .title {
        font-size: 16px;
    }
    .by{
        font-size: 14px;
    }
    .author{
        font-size: 15px;
    }
    .box{
        height: 290px;
    }
}
@media (min-width: 1100px) {
    .title {
        font-size: 18px;
    }
    .by{
        font-size: 15px;
    }
    .author{
        font-size: 15px;
    }
    .box{
        height: 400px;
    }
}