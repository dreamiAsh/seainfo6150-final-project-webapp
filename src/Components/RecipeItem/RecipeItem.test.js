import React from "react";
import { render, fireEvent } from "@testing-library/react";
import RecipeItem from "./RecipeItem.jsx";

it("renders the RecipeItem component correctly", () => {
  const recipe = {
      title: "a",
      imageId: "b",
      author: "c",
      authorInfo: "d",
      reviews: "a",
      likes: "b",
      shortText: "c",
      ingredient: "d",
      step1: "a",
      step2: "b"
  }
  const { container } = render(<RecipeItem recipe={recipe}/>);
  expect(container).toMatchSnapshot();
});