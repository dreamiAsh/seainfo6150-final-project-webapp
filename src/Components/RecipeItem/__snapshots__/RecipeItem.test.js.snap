// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the RecipeItem component correctly 1`] = `
<div>
  <a
    class="box"
    href="/recipe/a/b/c/d/a/b/c/d/a/b"
  >
    <div
      class="imgBox"
    >
      <img
        alt="recipe image"
        class="img"
      />
    </div>
    <p
      class="title"
    >
      a
    </p>
    <p
      class="by"
    >
      By 
      <span
        class="author"
      >
        c
      </span>
    </p>
  </a>
</div>
`;
